### ISSUE TITLE
Catch Failed self-updates

### ISSUE THREAD

**Issue by DirtyCajunRice** [pyouroboros/ouroboros#230]
Error:
```
2019-02-20 10:00:21 : ERROR : base : Job "Cron container update for tcp://10.0.10.140:2376 (trigger: cron[month='*', day='*', day_of_week='*', hour='*', minute='0'], next run at: 2019-02-20 11:00:00 CST)" raised an exception
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/docker/api/client.py", line 256, in _raise_for_status
    response.raise_for_status()
  File "/usr/local/lib/python3.7/site-packages/requests/models.py", line 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 409 Client Error: Conflict for url: http://10.0.10.140:2376/v1.35/containers/create?name=ouroboros

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/apscheduler/executors/base.py", line 125, in run_job
    retval = job.func(*job.args, **job.kwargs)
  File "/usr/local/lib/python3.7/site-packages/pyouroboros/dockerclient.py", line 297, in update
    self.update_self(old_container=container, new_image=latest_image, count=1)
  File "/usr/local/lib/python3.7/site-packages/pyouroboros/dockerclient.py", line 345, in update_self
    me_created = self.client.api.create_container(**new_config)
  File "/usr/local/lib/python3.7/site-packages/docker/api/container.py", line 427, in create_container
    return self.create_container_from_config(config, name)
  File "/usr/local/lib/python3.7/site-packages/docker/api/container.py", line 438, in create_container_from_config
    return self._result(res, True)
  File "/usr/local/lib/python3.7/site-packages/docker/api/client.py", line 262, in _result
    self._raise_for_status(response)
  File "/usr/local/lib/python3.7/site-packages/docker/api/client.py", line 258, in _raise_for_status
    raise create_api_error_from_http_exception(e)
  File "/usr/local/lib/python3.7/site-packages/docker/errors.py", line 31, in create_api_error_from_http_exception
    raise cls(e, response=response, explanation=explanation)
docker.errors.APIError: 409 Client Error: Conflict ("Conflict. The container name "/ouroboros" is already in use by container "b97cabb3a4b71f9e335252a3790b43b95a55c9aa1f6b52273470a119bf1ee30b". You have to remove (or rename) that container to be able to reuse that name.")
```

**Comment by circa10a** [comment]
Fixed in #238

**Comment by DirtyCajunRice** [comment]
The issue is when scheduling is set by cron, it does not run immediately. It instead runs on the next scheduled cron. So if you only have it run once a day, It will not catch itself within the 30 seconds. This also adds a condition where if it self updates, it will not update the rest of the containers that it had not updated yet. This can be fixed by adding a job like run once prior to adding the scheduler job for cron.
